### PR TITLE
Update WCPay banners for WooPay

### PR DESCRIPTION
**Pull request by alefesouza**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Update WCPay promotions on WooPay eligible countries to show WooPay references.

This change is related to 2020-gh-Automattic/woopay

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Follow 17784-gh-Automattic/woocommerce.com testing instructions.
2. Update [this line](https://github.com/woocommerce/woocommerce/blob/5efff2c8a28556f7d61e058cb483d8616421a5a9/plugins/woocommerce/src/Internal/Admin/WCPayPromotion/WCPayPromotionDataSourcePoller.php#L17) to `https://woocommerce.test/wp-json/wccom/payment-gateway-suggestions/1.0/payment-method/promotions.json`, or `http://host.docker.internal/wp-json/wccom/payment-gateway-suggestions/1.0/payment-method/promotions.json` if you are using Docker.
3. Comment out [this line](https://github.com/woocommerce/woocommerce/blob/9fec692865cbdcf61f011ab5d292555f694c28a8/plugins/woocommerce/src/Admin/DataSourcePoller.php#L111) and [this line](https://github.com/woocommerce/woocommerce/blob/9fec692865cbdcf61f011ab5d292555f694c28a8/plugins/woocommerce/src/Admin/DataSourcePoller.php#L115) to ignore cache.
4. Set your WooCommerce store country to US.
5. With WCPay uninstalled (you can just rename the plugin folder) and all payment methods disabled, access `WooCommerce > Overview > Set up payments`, it should show this banner with WooPay references

Screenshot 2023-08-07 at 3 03 03 PM

6. Access `WooCommerce > Settings > Payments` page, it should show a description mentioning WooPay.

Screenshot 2023-08-07 at 1 57 27 PM

7. Change the store country to a country that supports WCPay but not WooPay (ex. Canada).
8. Reload the `Set up payments` page, it should not mention WooPay.

Screenshot 2023-08-07 at 3 02 22 PM

9. Reload the Payments settings page, it should not mention WooPay.

Screenshot 2023-08-07 at 2 57 51 PM

10. Install WCPay but deactivate it.
11. Change your store country back to US.
12. Access `WooCommerce > Overview`.
13. The `Set up payments` section should reference WooPay merchant terms.

Screenshot 2023-08-07 at 2 55 04 PM

14. Change the store country to a country that supports WCPay but not WooPay (ex. Canada).
15. Reload the `WooCommerce > Overview` page.
16. It should not reference WooPay merchant terms.

Screenshot 2023-08-07 at 2 58 06 PM

17. Change the country to US.
17. Activate WCPay but disable it as a payment gateway on the `WooCommerce > Settings > Payments` page.
18. Reload the page.
19. The promotion banner should mention WooPay merchant terms.

Screenshot 2023-08-07 at 3 56 56 PM

21. Change the store country to a country that supports WCPay but not WooPay (ex. Canada).
22. Reload the `WooCommerce > Settings > Payments` page.
23. The promotion banner should not mention WooPay merchant terms.

Screenshot 2023-08-07 at 3 46 18 PM

24. WooPay should never be mentioned on old WooCommerce versions even if it's on a eligible country, you can change the WC version [here](https://github.com/woocommerce/woocommerce/blob/5efff2c8a28556f7d61e058cb483d8616421a5a9/plugins/woocommerce/woocommerce.php#L6) to an older version and repeat the eligible steps mentioned here and it should act as it was ineligible.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

**Comment by alefesouza** [comment]
@pierorocca @elizaan36 pinging as this is a visual changing PR.

**Comment by pierorocca** [comment]
I'm afk today with limited access. Would be appreciated if @elizaan36 could demo this.

File: plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/WooCommercePayments.php [hunk 1]
```diff
@@ -73,6 +76,17 @@ public function get_action_label() {
 	 * @return string
 	 */
 	public function get_additional_info() {
+		$data               = WCPayPromotionDataSourcePoller::get_instance()->get_specs_from_data_sources();
```
Comment (moon0326) [inline plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/WooCommercePayments.php]: The logic is somewhat duplicate of [getPromotions()](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Internal/Admin/WCPayPromotion/Init.php#L124). It's also bypassing [woocommerce_show_marketplace_suggestion](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Internal/Admin/WCPayPromotion/Init.php#L155) check.

What do you think about adding a method (e.g., isWooPayEligible) in the class?

File: plugins/woocommerce/src/Internal/Admin/Settings.php [hunk 2]
```diff
@@ -236,6 +239,13 @@ public function add_component_settings( $settings ) {
 
 		$settings['features'] = $this->get_features();
 
+		$data           = WCPayPromotionDataSourcePoller::get_instance()->get_specs_from_data_sources();
```
Comment (moon0326) [inline plugins/woocommerce/src/Internal/Admin/Settings.php]: The logic is somewhat duplicate of [getPromotions()](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Internal/Admin/WCPayPromotion/Init.php#L124)

What do you think about adding a method (e.g., isWooPayEligible) in the class?

File: plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/WooCommercePayments.php (hunk 1 above)
Comment (alefesouza) [reply plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/WooCommercePayments.php]: Good catch, fixed on https://github.com/woocommerce/woocommerce/pull/39596/commits/6ddc26468c6dd5215d7d1851328c15d8a8830ed1.

File: plugins/woocommerce/src/Internal/Admin/Settings.php (hunk 2 above)
Comment (alefesouza) [reply plugins/woocommerce/src/Internal/Admin/Settings.php]: Fixed on https://github.com/woocommerce/woocommerce/pull/39596/commits/6ddc26468c6dd5215d7d1851328c15d8a8830ed1.

**Review by moon0326 — APPROVED** [review]
LGTM and everything tested well 👍 🚀

**Review by hsingyuc — APPROVED** [review]
@alefesouza Nice work! Tests are as expected and LGTM!

**Pull request merged**